backend/fms_core/migrations/0016_v3_1_0.py

```diff
@@ -64,7 +44,7 @@ def create_lineage_from_extracted_and_revisions(apps, schema_editor):
     extraction_protocol = protocol_model.objects.get(name="Extraction")
     extracted_samples_info = {}
 
-    for revision in revision_model.objects.filter(comment="Imported extracted samples from template."):
+    for revision in revision_model.objects.filter(comment="Imported extracted samples from template.").iterator():
```

Comment:
your iterator() iterates on a queryset of 1... this would be more useful on the queryset iterating on the version_model.